File: mobile/lib/features/vehiculos/data/models/vehiculo_model.dart
import '../../domain/entities/vehiculo_entity.dart';

class VehiculoModel extends VehiculoEntity {
  const VehiculoModel({
    required super.id,
    required super.usuarioId,
    required super.marca,
    required super.modelo,
    required super.year,
    required super.placa,
    super.color,
    super.tipo,
    super.urlFoto,
    required super.estado,
    required super.fechaCreacion,
  });

  factory VehiculoModel.fromJson(Map<String, dynamic> json) => VehiculoModel(
        id: json['id'],
        usuarioId: json['usuario_id'],
        marca: json['marca'],
        modelo: json['modelo'],
        year: json['year'],
        placa: json['placa'],
        color: json['color'],
        tipo: json['tipo'],
        urlFoto: json['url_foto'],
        estado: json['estado'] ?? true,
        fechaCreacion: DateTime.parse(json['fecha_creacion']),
      );

  Map<String, dynamic> toJson() => {
        'marca': marca,
        'modelo': modelo,
        'year': year,
        'placa': placa,
        'color': color,
        'tipo': tipo,
        'url_foto': urlFoto,
        'estado': estado,
      };
}

File: Backend/app/models/vehiculo_model.py
from sqlalchemy import Column, Integer, String, Boolean, TIMESTAMP, ForeignKey, func
from sqlalchemy.orm import relationship
from app.core.db import Base


class Vehiculo(Base):
    __tablename__ = "vehiculos"

    id = Column(Integer, primary_key=True, index=True)
    usuario_id = Column(Integer, ForeignKey("usuario.id", ondelete="CASCADE"), nullable=False)
    marca = Column(String(100), nullable=False)
    modelo = Column(String(100), nullable=False)
    year = Column(Integer, nullable=False)
    placa = Column(String(20), unique=True, nullable=False)
    color = Column(String(50), nullable=True)
    tipo = Column(String(50), nullable=True)       # sedan, suv, pickup, moto, etc.
    url_foto = Column(String(255), nullable=True)
    estado = Column(Boolean, default=True)
    fecha_creacion = Column(TIMESTAMP, server_default=func.now())

    usuario = relationship("Usuario", back_populates="vehiculos")
    incidentes = relationship("Incidente", back_populates="vehiculo", cascade="all, delete-orphan")